core/src/main/java/tc/oc/pgm/filters/FilterMatchModule.java

```diff
@@ -198,6 +198,11 @@ private <F extends Filterable<?>> void register(
           "Filter " + filter + " doesn't respond to " + scope.getSimpleName() + " scope.");
     }
 
+    if (filter.getRelevantEvents().isEmpty()) {
+      // In some cases filters can be dynamic but doesn't need event invalidation
+      return;
+    }
+
```

Comment:
A filter with no events may still have listeners, particularily, monostable filters.

We should avoid listening for *event* changes on these, but the listeners should probably still be registered.